src/tagmanager/tm_workspace.c

```diff
@@ -681,56 +684,261 @@ GPtrArray *tm_workspace_find(const char *name, const char *scope, TMTagType type
 }
 
 
-static void fill_find_tags_array_prefix(GPtrArray *dst, const GPtrArray *src,
-	const char *name, TMParserType lang, guint max_num)
+gboolean tm_workspace_is_autocomplete_tag(TMTag *tag,
+	TMSourceFile *current_file,
+	guint current_line,
+	const gchar *current_scope)
 {
-	TMTag **tag, *last = NULL;
-	guint i, count, num;
+	TMParserType lang = current_file ? current_file->lang : TM_PARSER_NONE;
+
+	/* ignore local variables from other files/functions or after current line */
+	gboolean valid = !(tag->type & tm_tag_local_var_t) ||
+		(current_file == tag->file &&
+		 current_line >= tag->line &&
```

Comment:
Out of curiosity, why did you drop the idea of making this language specific? I think @elextr had a good example why this might not work so well for python.

Reply:
Because I think that even though it's syntactically valid, such cases are rare enough to justify seeing many more (invalid) local variables defined after the current line. But this is something that will have to be tested once Python local variables are enabled.

Reply:
OK